crates/sui-framework/deps/move-stdlib/sources/u64.move

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+/// Basic math for nicer programmability
+module std::u64 {
+    use std::ascii;
+    use std::option::{Self, Option};
+    use std::string;
+    use std::vector;
+
+    /// Largest possible u64 value
+    const MAX: u64 = 0;
+
+    /// Attempting to perform an invalid operation (e.g., mean) on an empty vector
+    const EVectorEmpty: u64 = 0;
+
+    /// Return the larger of `x` and `y`
+    public fun max(x: u64, y: u64): u64 {
+        if (x > y) {
+            x
+        } else {
+            y
+        }
+    }
+
+    /// Return the smaller of `x` and `y`
+    public fun min(x: u64, y: u64): u64 {
+        if (x < y) {
+            x
+        } else {
+            y
+        }
+    }
+
+    /// Return the value of a base raised to a power.
+    /// This function is not overflow-safe and will abort with an overflow error if `base^exponent` > `MAX`
+    public fun pow(base: u64, exponent: u8): u64 {
+        let res = 1;
+        while (exponent >= 1) {
+            if (exponent % 2 == 0) {
+                base = base * base;
+                exponent = exponent / 2;
+            } else {
+                res = res * base;
+                exponent = exponent - 1;
+            }
+        };
+
+        res
+    }
+
+    /// Return the mean of x and y, with the remainder truncated.
+    /// This function is overflow-safe and cannot abort
+    public fun mean(x: u64, y: u64): u64 {
```

Comment:
If there was a `min_max` function, you could define `min`, `max` and `mean` in terms of it:

```move
public fun min_max(x: u64, y: u64): (u64, u64) {
  if (x < y) {
    (x, y)
  } else {
    (y, x)
  }
}

public fun min(x: u64, y: u64): u64 {
  let (m, _) = min_max(x, y);
  m
}

public fun max(x: u64, y: u64): u64 {
  let (_, m) = min_max(x, y);
  m
}

public fun mean(x: u64, y: u64): u64 {
  let (lo, hi) = min_max(x, y);
  lo + (hi - lo) / 2
}
```